cppcache/src/ClientProxyMembershipID.cpp

```diff
@@ -201,65 +193,60 @@ void ClientProxyMembershipID::toData(DataOutput&) const {
+  input.readBytesOnly(hostAddress, length);
+  auto hostPort = input.readInt32();
   auto hostname =
       std::dynamic_pointer_cast<CacheableString>(input.readObject());
-  auto splitbrain = input.read();   // splitbrain
-  auto dcport = input.readInt32();  // port
-  auto vPID = input.readInt32();    // pid
-  auto vmKind = input.read();       // vmkind
+  auto splitbrain = input.read();
+  auto dcport = input.readInt32();
+  auto vPID = input.readInt32();
+  auto vmKind = input.read();
   auto aStringArray = CacheableStringArray::create();
   aStringArray->fromData(input);
   auto dsName = std::dynamic_pointer_cast<CacheableString>(input.readObject());
   auto uniqueTag =
       std::dynamic_pointer_cast<CacheableString>(input.readObject());
   auto durableClientId =
       std::dynamic_pointer_cast<CacheableString>(input.readObject());
-  auto durableClntTimeOut =
-      std::chrono::seconds(input.readInt32());  // durable client timeout
+  auto durableClientTimeOut = std::chrono::seconds(input.readInt32());
   int32_t vmViewId = 0;
   readVersion(splitbrain, input);
 
-  initHostAddressVector(hostAddr, len);
+  initHostAddressVector(hostAddress, length);
 
   if (vmKind != ClientProxyMembershipID::LONER_DM_TYPE) {
     vmViewId = std::stoi(uniqueTag->value());
     initObjectVars(hostname->value().c_str(), hostPort,
-                   durableClientId->value().c_str(), durableClntTimeOut, dcport,
-                   vPID, vmKind, splitbrain, dsName->value().c_str(), nullptr,
-                   vmViewId);
+                   durableClientId->value().c_str(), durableClientTimeOut,
+                   dcport, vPID, vmKind, splitbrain, dsName->value().c_str(),
+                   nullptr, vmViewId);
   } else {
     // initialize the object
     initObjectVars(hostname->value().c_str(), hostPort,
-                   durableClientId->value().c_str(), durableClntTimeOut, dcport,
-                   vPID, vmKind, splitbrain, dsName->value().c_str(),
+                   durableClientId->value().c_str(), durableClientTimeOut,
+                   dcport, vPID, vmKind, splitbrain, dsName->value().c_str(),
                    uniqueTag->value().c_str(), 0);
   }
-  delete[] hostAddr;
+  delete[] hostAddress;
   readAdditionalData(input);
 }
 
 Serializable* ClientProxyMembershipID::readEssentialData(DataInput& input) {
-  uint8_t* hostAddr;
-  int32_t len, hostPort, vmViewId = 0;
+  uint8_t* hostAddress;
+  int32_t length, hostPort, vmViewId = 0;
   std::shared_ptr<CacheableString> hostname, dsName, uniqueTag, vmViewIdstr;
 
-  len = input.readArrayLength();  // inetaddress len
-  /* adongre - Coverity II
-   * CID 29183: Out-of-bounds access (OVERRUN_DYNAMIC)
-   */
-  hostAddr = new uint8_t[len];
-  input.readBytesOnly(hostAddr, len);  // inetaddress
-  hostPort = input.readInt32();        // port
-  // TODO: RVV get the host name from
+  length = input.readArrayLength();
+  hostAddress = new uint8_t[length];
```

Comment:
Can we use a `std::vector` to avoid the `delete` and potential leak?

Reply:
I got a little stuck.  I was thinking `hostdress = std::vector<uint8_t>(input.read());` but that's not quite it.  I think I'm confused because I'm not sure how to get data from `input` into a new vector.

Reply:
This works:
```c++
  auto length = input.readArrayLength();
  auto hostAddress = std::vector<uint8_t>(length);
  input.readBytesOnly(hostAddress.data(), hostAddress.size());
```

or, use the method that reads the array length and bytes in a single call.
```c++
  auto hostAddress = input.readByteArray();
```
Though, this is a `std::vector<int8_t>` (signed byte like Java) now so you will need a `static_cast<>` later for the call to `initHostAddressVector`.

But you will notice that `initHostAddressVector` just copies it to the `m_hostAddr` which is a `std::vector<uint8_t>`, so maybe there is some more optimizations to be made... Rabbit hole now...

Reply:
for some reason `input.readBytesOnly(hostAddress.data(), hostAddress.size());` never dawned on me as an option 😞 
The latter one I started, but saw the same issue you mentioned